pkg/csi/service/common/authmanager.go

```diff
@@ -0,0 +1,158 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/common/cns-lib/vsphere"
+	cnsvsphere "sigs.k8s.io/vsphere-csi-driver/pkg/common/cns-lib/vsphere"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+type AuthorizationService interface {
+
+	// GetDatastoreIgnoreMapForBlockVolumes returns a map which maps datastore url to datastore info which user does not have Datastore.FileManagement privilege
+	// returns an error if not able to retrieve the map
+	GetDatastoreIgnoreMapForBlockVolumes(ctx context.Context) (map[string]*cnsvsphere.DatastoreInfo, error)
+}
+
+type AuthManager struct {
+	// map the datastore url to datastore info which need to be ignored when invoking CNS CreateVolume API
+	datastoreIgnoreMapForBlockVolumes map[string]*cnsvsphere.DatastoreInfo
+	// this mutex is to make sure the update for datatoreIgnoreMap is mutually exclusive
+	mutex sync.Mutex
```

Comment:
Can we use a RWMutex? This way you dont need to block concurrent readers(callers of GetDatastoreIgnoreMapForBlockVolumes) getting the datastoreIgnoreMapForBlockVolumes.

Reply:
Done.